src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -547,12 +554,47 @@ def validate_hosts_and_networks(self, other_shares, data, schema_name, verrors,
 
             used_networks.add(network)
 
-        if not data["hosts"] and not data["networks"]:
-            if used_networks:
-                verrors.add(
-                    f"{schema_name}.networks",
-                    "Another NFS share already exports this dataset for some network"
-                )
+    @private
+    async def validate_share_path(self, other_shares, data, schema_name, verrors):
+        """
+        Checks new share path against existing
+        """
+        realPath = os.path.realpath(data["path"])
```

Comment:
Since this method is a coroutine (`async def validate_share_path`), performing file I/O will block the main event loop since `os.path.realpath` just calls `lstat` underneath the hood.